utils/reflector/src/main/java/org/robolectric/util/reflector/Reflector.java

```diff
@@ -39,6 +39,7 @@ public class Reflector {
   private static final AtomicInteger COUNTER = new AtomicInteger();
   private static final Map<Class<?>, Constructor<?>> cache = new ConcurrentHashMap<>();
   private static final Map<Class<?>, Object> staticReflectorCache = new ConcurrentHashMap<>();
+  private static final Map<Class<?>, ForType> forTypeAnnotationCache = new ConcurrentHashMap<>();
```

Comment:
Is it possible to go a step further and just have this be a `Map<Class<?>, Class<?>>`, where it maps the reflector class to the fortype class?

Reply:
It's a great point. Thank you.